### PR TITLE
Rename Jetpack Migration plugin to "Move to WordPress.com"

### LINKED ISSUE #28736: Make sure all user-facing Jetpack branding is removed
- [x] Make sure the naming is correct across the codebase
- [x] Ping Jetpack crew to update the slug once we decide it.

### PR DESCRIPTION
**Pull request by ouikhuan**
Fixes #28736 

## Proposed changes:

* Rename Jetpack Migration plugin to "Move to WordPress.com".
* Rename slug from `jetpack-migration` to `wpcom-migration`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

P2 discussion: p9dueE-6Cx-p2
## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

* jetpack build -> plugins -> migration and spin up your local WordPress.
* Use jurassic tube to make the site public
* Navigate to the plugin page main page, Installed plugin page and try to connect and disconnect Jetpack to see if the functionality is fine and the wording are correct.
* There shouldn't be anything like `Jetpack Migration` or slug like `jetpack-migration`.

### CODE REVIEW

File: projects/plugins/migration/composer.json
Comment (vishnugopal) [inline projects/plugins/migration/composer.json:62]: Think this needs to be changed by Jetpack Crew too as it references a repo.

File: projects/plugins/migration/package.json
Comment (vishnugopal) [inline projects/plugins/migration/package.json]: https://wordpress.com/ ..?

File: projects/plugins/migration/wpcom-migration.php
Comment (vishnugopal) [inline projects/plugins/migration/wpcom-migration.php]: Let's redefine this as https://wordpress.com/

**Review by vishnugopal — COMMENTED** [review]
I haven't reviewed this in detail, just left some comments from the code perspective.

File: projects/plugins/migration/composer.json
Comment (ouikhuan) [reply projects/plugins/migration/composer.json:62]: Yes, this is correct, once we have an approval on our team review, I'll communicate with Jetpack crew to help us makes the changes 👍

File: projects/plugins/migration/package.json
Comment (ouikhuan) [reply projects/plugins/migration/package.json]: Thanks for catching this, updated as suggestion!

File: projects/plugins/migration/wpcom-migration.php
Comment (ouikhuan) [reply projects/plugins/migration/wpcom-migration.php]: Updated, thanks!

**Review by jeherve — APPROVED** [review]
This looks good to me. Let's flip the switch! I've updated the related settings (mirror repo, beta server) so we should be good to go!

**Comment by ouikhuan** [comment]
Thanks for the help @jeherve !

**Pull request merged**
